### ISSUE TITLE
Inconsistent Time & Location in indi_lx200_10micron

### ISSUE THREAD

**Issue by airscottdenning** [mworion/MountWizzard4#93]
Hi Michel, I apologize for reporting this as a GitHub "issue." I don't have your contact info any other way.

I just [posted in the INDI forums about an inconsistency in time & location data between the indi_lx200_10micron driver and the 10Micron mount when using the MGPBox](https://indilib.org/forum/general/8391-inconsistent-time-location-in-indi-lx200-10micron-was-3-5-0-flip-failure.html). You had previously warned me about making certain there was only a single source of time & location data among the mount, GPS, driver, and imaging software. 

I have been unable to enforce this except by completely disabling the MGPBox and setting time and location to "Kstars updates all devices" in Ekos. This is MUCH better for me than allowing the mount and INDI to fight over mount position, which leads to horrible tracking in DEC. 

But I wanted to ask you for advice. Do YOU image with your 10Micron in KStars/Ekos? How do you *avoid* conflicts between the mount/MGPBox and INDI driver?

I'm pasting the text of my INDO Forum post below, but hopefully it's easy for you to see the images on their forum. 

Please let me know if you'd rather I emailed you than reporting issues unrelated to MW4 here.

Scott

========

Inconsistent Time & Location in indi_lx200_10micron (was 3.5.0 Flip Failure) #64490

There appears to be a time and location conflict in the indi_lx200_10micron driver when the mount is updated from its own GPS (MGPBox), resulting in a range of bad behavior including failed meridian flips, bad pointing, and horrific tracking when the mount and INDI and KStars fight over precise geographic coordinates and time.

I build a mount model in separate software (ModelWizzard 4) that leads to incredibly precise pointing without KStars and 2-axis tracking using built-in encoders. But when doing production imaging in KStars I get horrible arcMINUTE-scale star trails as the mount and software fight over the time & location (see example images attached below).

The time mismatch between GPS/10Micron and Kstars frequently leads to failed meridian flips as the mount reaches a safety stop before KStars realizes it's time to flip.

Another symptom is that the reported geographic site position in the INDI driver doesn't retain arcseconds in geographic coordinates (for consistency with LX200?) but the GPS-reported coordinates in the mount keypad does. This was reported by fenriques in this thread. It may be that the arcseconds numbers are just not being reported in the GUI, but the rest of the symptoms suggest that the actual geographic coordinates are being *truncated* in the driver, resulting in the DEC thrashing I see in my subs (unless I just ignore the mount coordinates).

As Andrew points out in this thread , I can get pretty good results by just ignoring all the time & location info in the mount and GPS. To do this I have to turn off the MGPBox syncing in the mount hand controller set the geographic coordinates by hand, and set "KStars updates mount" in the Ekos settings. But I get nearly PERFECT results in TheSkyX by just trusting the mount to do its own thing and not trying to second-guess it in software.

This is a tough issue to diagnose because it involves the mount computer as well as the INDI driver and KStars/Ekos. I can save logs but not sure which logs are involved. Maybe I need to collect a mount log from the 10micron computer and report it to the vendor? But pretty sure this is cross-talk between INDI and 10Micron so hard to know what info to provide to whom.

Attached are two unguided 180-second subs of M74. Polar alignment is identical in these two unguided subs (well under 1 arc minute PA error). The first (with coordinates overlaid) shows INDI and the mount fighting over the time and location. The second shows good 2-axis tracking with all GPS info disabled. It gets even better (eccentricity under 0.3) when I use other software like TheSkyX that just leaves the mount alone to do its own thing. Everything I report here is for KStars 3.5.0 (regular ppa, not nightly) on ubuntu 20.04.1 running on R Pi 4.

This post is following up on three other related threads which I suspect are due to the same issue:

https://indilib.org/forum/general/8328-3-5-0-flip-failure-10micron.html 

https://indilib.org/forum/general/7928-why-is-the-utc-time-in-kstars-ekos-different-from-my-mount.html?start=0 

https://indilib.org/forum/general/7928-why-is-the-utc-time-in-kstars-ekos-different-from-my-mount.html?start=0

**Comment by mworion** [comment]
Scott,

no problem. Ferrante already asked similar topics and I passed him to Hans Lambermont, who wrote the 10micron driver. He is a very nice guy and also very helpful. I think the best way is to add an option to the driver.

I will give him anyway a short notice that there might be some issues.

My personal experience so far: I do imaging with KStars/EKOS for 3 Years now. I also have a MGBox attached an used for boot time. I had some issues sometimes of unround stars (eccentricity of 0.6), but no elongated stars like you describe. But I could follow the changes the driver does, so the behaviour  is not correct and should be changed. But i will have a closer look to it the next days.

Michel

**Comment by mworion** [comment]
Hi Scott,

did you check the settings in KStars/Ekos about which side updates the other ? I had set it to mount updates KStars. Does it work out for you ?

Michel

**Comment by airscottdenning** [comment]
Hi Michel,

I have tried all the possible combinations in KStars/Ekos: I started with mount updates KStars. Then GPS updates KStars. Then KStars updates mount. Finally (at Jasem’s suggestion) nobody updates anything (uncheck both boxes).

Then I tried just switching to Windows and ASCOM and ModelCreator and TheSkyX, without *any* INDI drivers or Ekos in the mix.

But my mount still does these weird mini-slews in DEC that leave trailed stars.

I think there’s something wrong with my mount. I have submitted logs to 10Micron and I’m waiting to host a remote access session to help diagnose it.

Thank you very much for your suggestions!

Scott
